can't locate @ember-data/record-data/-private referred to in module rules

I'm running tests in ember-try `embroider-safe` and we're getting the following error: 


```
Build Error (WaitForTrees)

can't locate @ember-data/record-data/-private referred to in module rules:{
  "dependsOnModules": [
    "@ember-data/record-data/-private"
  ]
}
```

I've done some debugging and I've found the error that is being thrown in the `resolver.js`

```
'Cannot find module '@ember-data/record-data/-private' from '/private/var/folders/r5/bw5frvnn5k794hk62w3vp7km0000gn/T/embroider/2daa8e/node_modules/.pnpm/@ember-data+store@4.8.4_k6kp2ds5o4a3bn46aja3did5ee/node_modules/@ember-data/store''
```

I had a look at the file on disk and it looks like it's right, there was no sibling to `@ember-data/store` for `record-data` and normal node resolution would be expected to fail.

--- a/packages/compat/src/addon-dependency-rules/ember-data.ts
+++ b/packages/compat/src/addon-dependency-rules/ember-data.ts
@@ -3,6 +3,7 @@ import { PackageRules } from '..';
 let rules: PackageRules[] = [
   {
     package: '@ember-data/store',
+    semverRange: '<=4.11.0',
     addonModules: {
       '-private.js': {
         dependsOnModules: ['@ember-data/record-data/-private'],

--- a/packages/compat/src/compat-adapters/ember-data.ts
+++ b/packages/compat/src/compat-adapters/ember-data.ts
@@ -3,6 +3,7 @@ import { join } from 'path';
 import { Memoize } from 'typescript-memoize';
 import { Node } from 'broccoli-node-api';
 import { sync as resolveSync } from 'resolve';
+import semver from 'semver';
 
 export class EmberDataBase extends V1Addon {
   // May of the ember-data packages use rollup to try to hide their internal
@@ -13,6 +14,10 @@ export class EmberDataBase extends V1Addon {
   // behavior is correct.
   customizes(...names: string[]) {
     return super.customizes(...names.filter(n => n !== 'treeForAddon'));
+  }
+
+  static shouldApplyAdapter(addonInstance: any) {
+    return semver.lt(addonInstance.pkg.version, '4.11.1');
   }
 }
 

